### DISCUSSION TITLE
Ignore all hosts except one

### DISCUSSION THREAD

**Discussion by qol3m** [mitmproxy/mitmproxy#6497] [Q&A]
I want to spy on an android app, but only one specific domain (the one with the API). The other domains (googleapis stuff) need to be passed through, because their certificate pinning seems to be built into android, rather than the app.

And of course seeing all the unnecessary requests is annoying.

./mitmproxy --set block_global=false --ignore '^(?!ab-cd\.efg\.com)(?!mitmproxy\.org)'

does not work. I tried to escape the - , too. The app just does not work then. No requests show up in mitmproxy. What now?

Thanks

**Comment by dstd** [comment]
It's because mitmproxy takes a list of a domain and an IP and matches the expression to ANY of them. Your expression matches the IP so that the requests are ignored.
